spacy/tests/pipeline/test_spancat.py

```diff
@@ -372,31 +372,56 @@ def test_overfitting_IO_overlapping():
 
 
 def test_zero_suggestions():
-    # Test with a suggester that returns 0 suggestions
+    # Test with a suggester that can return 0 suggestions
 
-    @registry.misc("test_zero_suggester")
-    def make_zero_suggester():
-        def zero_suggester(docs, *, ops=None):
+    @registry.misc("test_mixed_zero_suggester")
+    def make_mixed_zero_suggester():
+        def mixed_zero_suggester(docs, *, ops=None):
             if ops is None:
                 ops = get_current_ops()
-            return Ragged(
-                ops.xp.zeros((0, 0), dtype="i"), ops.xp.zeros((len(docs),), dtype="i")
-            )
-
-        return zero_suggester
+            spans = []
+            lengths = []
+            for doc in docs:
+                if len(doc) > 0 and len(doc) % 2 == 0:
+                    spans.append((0, 1))
+                    lengths.append(1)
+                else:
+                    lengths.append(0)
```

Comment:
The test here is passing, but I'm having trouble understanding why having the `spans` and `lengths` being different lengths works. Is it intentional that they end up different lengths? Should a Ragged where the input array and the lengths are different lengths be valid?

Reply:
`lengths` is the number of spans per doc. Here, some docs have 1 span and some docs have 0 spans.

Reply:
Got it, that makes sense. I had some trouble following the difference between span lengths, number of spans per doc, and indices in used in extracting spans.